File: temp_files/update_ui.dart
// @dart=2.12
import 'dart:io';

void main() {
  var file = File('lib/src/app_common/login/select_module.dart');
  var content = file.readAsStringSync();
  
  content = content.replaceAll(
    'const Color(0xff1F59DE)),',
    'const Color(0xff1F59DE), hasSparkle: true),'
  );
  content = content.replaceAll(
    'const Color(0xff7F15D1)),',
    'const Color(0xff7F15D1), hasSparkle: true),'
  );
  content = content.replaceAll(
    'Color.fromARGB(255, 184, 8, 178)),',
    'Color.fromARGB(255, 184, 8, 178), hasSparkle: true),'
  );

  content = content.replaceAll(
    '''Widget _buildItem(
      String icon, String title, Function() onTap, Color bgColor) {''',
    '''Widget _buildItem(
      String icon, String title, Function() onTap, Color bgColor, {bool hasSparkle = false}) {'''
  );

  var originalCardChild = '''          child: Container(
            padding: EdgeInsets.symmetric(
                vertical: isLargeScreenSize ? 20 : 16,
                horizontal: isLargeScreenSize ? 20 : 12),
            child: Row(
              children: [
                Padding(
                    padding: const EdgeInsets.only(right: 16),
                    child: SvgPicture.asset(
                      icon,
                      color: Colors.white,
                    )),
                Expanded(
                    child: Text(
                  title ?? '',
                  style: const TextStyle(
                      fontSize: 16,
                      fontWeight: FontWeight.w400,
                      color: Colors.white),
                )),
                Padding(
                  padding: const EdgeInsets.only(left: 16),
                  child: SvgPicture.asset(
                    ImagesCommon.icBack,
                    color: Colors.white,
                  ),
                )
              ],
            ),
          ),''';

  var newCardChild = '''          clipBehavior: Clip.antiAlias,
          child: Stack(
            children: [
              Container(
                padding: EdgeInsets.symmetric(
                    vertical: isLargeScreenSize ? 20 : 16,
                    horizontal: isLargeScreenSize ? 20 : 12),
                child: Row(
                  children: [
                    Padding(
                        padding: const EdgeInsets.only(right: 16),
                        child: SvgPicture.asset(
                          icon,
                          color: Colors.white,
                        )),
                    Expanded(
                        child: Text(
                      title ?? '',
                      style: const TextStyle(
                          fontSize: 16,
                          fontWeight: FontWeight.w400,
                          color: Colors.white),
                    )),
                    Padding(
                      padding: const EdgeInsets.only(left: 16),
                      child: SvgPicture.asset(
                        ImagesCommon.icBack,
                        color: Colors.white,
                      ),
                    )
                  ],
                ),
              ),
              if (hasSparkle)
                const Positioned(
                  top: -5,
                  right: 10,
                  child: Text(
                    '✨⚡️',
                    style: TextStyle(fontSize: 26, shadows: [
                      Shadow(
                        blurRadius: 10.0,
                        color: Colors.yellow,
                        offset: Offset(0, 0),
                      ),
                    ]),
                  ),
                ),
            ],
          ),''';

  content = content.replaceAll(originalCardChild, newCardChild);
  
  file.writeAsStringSync(content);
}

File: temp_files/update_ui.py
import re
with open('lib/src/app_common/login/select_module.dart', 'r', encoding='utf-8') as f:
    text = f.read()

text = re.sub(
    r'(Widget _buildItem\([^)]+Color bgColor\)) \{',
    r'\1, {bool hasSparkle = false}) {',
    text
)

old_card = '''        child: Card(
          color: bgColor,
          child: Container('''
new_card = '''        child: Card(
          color: bgColor,
          clipBehavior: Clip.antiAlias,
          child: Stack(
            children: [
              Container('''
text = text.replace(old_card, new_card)

old_end = '''              ],
            ),
          ),
        ),
      ),
    );'''
new_end = '''              ],
            ),
          ),
          if (hasSparkle)
            const Positioned(
              top: 0,
              right: 15,
              child: Text(
                '✨⚡️',
                style: TextStyle(fontSize: 26, shadows: [
                  Shadow(
                    blurRadius: 10.0,
                    color: Colors.yellow,
                    offset: Offset(0, 0),
                  ),
                ]),
              ),
            ),
        ],
      ),
    ),
  ),
);'''
text = re.sub(r'              \],\s*\),\s*\),\s*\),\s*\),\s*\);', new_end, text)

with open('lib/src/app_common/login/select_module.dart', 'w', encoding='utf-8') as f:
    f.write(text)
